lib/services/elasticsearch.js

```diff
@@ -1109,6 +1137,35 @@ class ElasticSearch extends Service {
     return this._mexecute(esRequest, extracted.documents, extracted.rejected);
   }
 
+  /**
+   * Create or replace multiple documents at once.
+   * Does not adds additional metadata
+   *
+   * @param  {Request} request - Kuzzle API request
+   * @return {Promise}
+   */
+  mwriteDocuments(request) {
+    const
+      esRequest = initESRequest(request, ['consistency', 'refresh', 'timeout', 'fields']),
+      extracted = extractMDocuments(request, {});
+
+    esRequest.body = [];
+
+    let i; // NOSONAR
```

Comment:
Why don't you declare `i` in the `for` loop ?

Reply:
It's a performance issue on Node.js 6 if you didn't declare the let outside the loop (see [this](https://stackoverflow.com/questions/37792934/why-is-let-slower-than-var-in-a-for-loop-in-nodejs) )
Since this method will potentially handle thousands of documents it's better to use this tricks